apps/indexer/test/indexer/fetcher/celo_election_rewards_test.exs

```diff
@@ -0,0 +1,424 @@
+defmodule Indexer.Fetcher.CeloElectionRewardsTest do
+  use EthereumJSONRPC.Case, async: false
+  use Explorer.DataCase
+
+  import Ecto.Query
+  import Explorer.Celo.CacheHelper
+  import Explorer.Factory
+  import Mox
+
+  alias Explorer.Celo.ContractEvents.Election.ValidatorGroupVoteActivatedEvent
+  alias Explorer.Celo.ContractEvents.Validators.ValidatorEpochPaymentDistributedEvent
+  alias Explorer.Chain.{Address, Block, CeloElectionRewards}
+  alias Indexer.Fetcher.CeloElectionRewards, as: CeloElectionRewardsFetcher
+
+  @moduletag :capture_log
+
+  # MUST use global mode because we aren't guaranteed to get `start_supervised`'s pid back fast enough to `allow` it to
+  # use expectations and stubs from test's pid.
+  setup :set_mox_global
+
+  setup :verify_on_exit!
+
+  setup do
+    start_supervised!({Task.Supervisor, name: Indexer.TaskSupervisor})
+
+    # Need to always mock to allow consensus switches to happen on demand and protect from them happening when we don't
+    # want them to.
+    %{
+      json_rpc_named_arguments: [
+        transport: EthereumJSONRPC.Mox,
+        transport_options: [],
+        # Which one does not matter, so pick one
+        variant: EthereumJSONRPPC.Parity
+      ]
+    }
+  end
+
+  describe "async_fetch/1" do
+    setup [:save_voter_contract_events_and_start_fetcher]
+
+    test "with consensus block without reward", context do
+      CeloElectionRewardsFetcher.async_fetch([
+        %{
+          block_number: context.last_block_in_epoch_number,
+          block_timestamp: DateTime.utc_now()
+        }
+      ])
+
+      wait_for_results(fn ->
+        reward = Repo.one!(from(rewards in CeloElectionRewards))
+
+        assert reward.reward_type == "voter"
+      end)
+
+      # Terminates the process so it finishes all Ecto processes.
+      GenServer.stop(context.pid)
+    end
+  end
+
+  describe "init/2" do
+    test "buffers unindexed epoch blocks", %{
+      json_rpc_named_arguments: json_rpc_named_arguments
+    } do
+      block = insert(:block)
+      insert(:celo_pending_epoch_operations, block_number: block.number)
+
+      assert CeloElectionRewardsFetcher.init(
+               [],
+               fn block_number, acc -> [block_number | acc] end,
+               json_rpc_named_arguments
+             ) == [%{block_number: block.number, block_timestamp: block.timestamp}]
+    end
+
+    test "does not buffer blocks with fetched election rewards", %{
+      json_rpc_named_arguments: json_rpc_named_arguments
+    } do
+      block = insert(:block)
+      insert(:celo_pending_epoch_operations, block_number: block.number, election_rewards: false)
+
+      assert CeloElectionRewardsFetcher.init(
+               [],
+               fn block_number, acc -> [block_number | acc] end,
+               json_rpc_named_arguments
+             ) == []
+    end
+  end
+
+  describe "get_voter_rewards, when voter rewards are passed" do
```

Comment:
Would be great to actually test the `get_voter_rewards` method as it looks like you deleted `celo_voter_votes_test`.

Reply:
True but that test doesn't check any values except for the reward type. I thought you had some tests that check reward values in the previous version somewhere.